File: Questions/two_sum/Go/solution/ans.go
package solution

func TwoSum(nums []int, target int) []int {
	numMap := make(map[int]int)

	for i, num := range nums {
		numMap[num] = i
	}

	for i, num := range nums {
		remain := target - num
		if idx, ok := numMap[remain]; ok {
			if idx != i {
				return []int{idx, i}
			}
		}
	}
	return nil
}

File: Questions/two_sum/Py/ans.py
def twoSum(nums: list[int], target: int) -> list[int]:
    num_pos = {n: i for i, n in enumerate(nums)}

    for i in range(len(nums)):
        remain = target - nums[i]
        if remain in num_pos and num_pos[remain] != i:
            return [i, num_pos[remain]]
    return []
